manager/integration/tests/test_settings.py

```diff
@@ -183,26 +191,87 @@ def test_setting_toleration_extra():
+            "expect": [
+                {
+                    "key": "ex.com/foobar",
+                    "value": None,
+                    "operator": "Exists",
+                    "effect": "NoExecute"
+                },
+                {
+                    "key": "ex.com/foobar",
+                    "value": None,
+                    "operator": "Exists",
+                    "effect": "NoSchedule"
+                },
+            ],
+        },
+        {
+            "value": "node-role.kubernetes.io/controlplane=true:NoSchedule",
+            "expect": [
+                {
+                    "key": "node-role.kubernetes.io/controlplane",
+                    "value": "true",
+                    "operator": "Equal",
+                    "effect": "NoSchedule"
+                },
+            ],
+        },
+        {
+            "value": ":",
+            "expect": [
+                {
+                    "key": None,
+                    "value": None,
+                    "operator": "Exists",
+                    "effect": None,
+                },
+            ]
+        },
+        {
+            "value": "",
+            "expect": [],
+        },
+    ]
+
+    chk_removed_tolerations = []
+    for setting in settings:
+        client = get_longhorn_api_client()  # NOQA
+        taint_toleration = client.by_id_setting(SETTING_TAINT_TOLERATION)
+        updated = client.update(taint_toleration,
+                                value=setting["value"])
+        assert updated.value == setting["value"]
+
+        core_api = get_core_api_client()  # NOQA
+        node_count = len(client.list_node())
+        wait_for_toleration_update(core_api, apps_api, node_count,
+                                   setting["expect"], chk_removed_tolerations)
+        chk_removed_tolerations = setting["expect"]
+
+
+def wait_for_toleration_update(core_api, apps_api, count,  # NOQA
+                               expected_tolerations,
+                               chk_removed_tolerations=[]):
     updated = False
 
-    for i in range(RETRY_COUNTS):
+    for _ in range(RETRY_COUNTS):
         time.sleep(RETRY_INTERVAL_LONG)
-        updated = True
+        updated = False
 
         if not check_workload_update(core_api, apps_api, count):
             updated = False
             continue
 
         pod_list = core_api.list_namespaced_pod(LONGHORN_NAMESPACE).items
         for p in pod_list:
-            if p.status.phase != "Running" or \
-                    not check_tolerations_set(p.spec.tolerations,
-                                              set_tolerations):
-                updated = False
+            if p.status.phase == "Running" \
+                and check_tolerations_set(p.spec.tolerations,
+                                          expected_tolerations,
+                                          chk_removed_tolerations):
+                updated = True
```

Comment:
I think there is an issue here. The expectation is that all pods in `longhorn-system` namespace must be updated. However, the new code is reporting ok if there is a pod that was updated. How do you think?

Reply:
You are correct. Addressed.